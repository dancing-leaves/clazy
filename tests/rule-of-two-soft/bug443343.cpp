struct Test
{
    Test() {}
    ~Test() { }

    Test(const Test &) {}
    Test& operator=(const Test&) = default;
};

void test()
{
    Test t;
    Test t2;
    t = t2; // OK, the developer explicitly says he wants the default copy-assign op
}

struct Test2
{
    Test2() {}
    ~Test2() { }

    Test2(const Test2 &) {}
};

void test2()
{
    Test2 t;
    Test2 t2;
    t = t2; // Warn
}
