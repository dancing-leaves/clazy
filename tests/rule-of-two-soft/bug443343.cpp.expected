rule-of-two-soft/bug443343.cpp:29:5: warning: Using assign operator but class Test2 has copy-ctor but no assign operator [-Wclazy-rule-of-two-soft]
